gradoop-examples/src/main/java/org/gradoop/benchmark/cypher/CypherBenchmark.java

```diff
@@ -0,0 +1,263 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.benchmark.cypher;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.io.FileUtils;
+import org.apache.flink.api.common.ProgramDescription;
+import org.apache.flink.api.java.ExecutionEnvironment;
+import org.apache.hadoop.conf.Configuration;
+import org.gradoop.benchmark.subgraph.SubgraphBenchmark;
+import org.gradoop.examples.AbstractRunner;
+import org.gradoop.flink.io.api.DataSink;
+import org.gradoop.flink.io.api.DataSource;
+import org.gradoop.flink.io.impl.csv.indexed.IndexedCSVDataSource;
+import org.gradoop.flink.io.impl.json.JSONDataSink;
+import org.gradoop.flink.model.api.epgm.GraphCollection;
+import org.gradoop.flink.model.api.epgm.LogicalGraph;
+import org.gradoop.flink.model.impl.operators.matching.common.statistics.GraphStatistics;
+import org.gradoop.flink.model.impl.operators.matching.common.statistics.GraphStatisticsHDFSReader;
+import org.gradoop.flink.util.GradoopFlinkConfig;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A dedicated program for parametrized graph cypher benchmark.
+ */
+public class CypherBenchmark extends AbstractRunner implements ProgramDescription {
+  /**
+   * Option to declare path to input graph
+   */
+  private static final String OPTION_INPUT_PATH = "i";
+  /**
+   * Option to declare path to output graph
+   */
+  private static final String OPTION_OUTPUT_PATH = "o";
+  /**
+   * Option do declare path to statistics
+   */
+  private static final String OPTION_STATISTICS_PATH = "s";
+  /**
+   * Option to declare path to statistics csv file
+   */
+  private static final String OPTION_CSV_PATH = "c";
+  /**
+   * Option to declare verification
+   */
+  private static final String OPTION_QUERY = "q";
+  /**
+   * Option for used first name in query.
+   */
+  private static final String OPTION_FIRST_NAME = "n";
+  /**
+   * Used input path
+   */
+  private static String INPUT_PATH;
+  /**
+   * Used output path
+   */
+  private static String OUTPUT_PATH;
+  /**
+   * Used to indicate if statistics are used
+   */
+  private static boolean HAS_STATISTICS;
+  /**
+   * Used for statistics path
+   */
+  private static String STATISTICS_INPUT_PATH;
+  /**
+   * Used csv path
+   */
+  private static String CSV_PATH;
+  /**
+   * Used query
+   */
+  private static String QUERY;
+  /**
+   * Used first name for query (q1,q2,q3)
+   */
+  private static String FIRST_NAME;
+
+  static {
+    OPTIONS.addOption(OPTION_INPUT_PATH, "input", true,
+      "Path to source files.");
```

Comment:
add in the description (also in JavaDoc above of `OPTION_INPUT_PATH`) that only a IndexedCsv is valid.

Reply:
did we really need this static code block? I'm not much of a fan of this pattern